chore(blog): summary of the asyncapi mentorship program 2024 (#4292)
asyncapi/website: markdown/blog/2024-mentorship-program-summary.md

---
title: "AsyncAPI Mentorship Program 2024 - Wrap Up"
date: 2025-08-01T06:00:00+01:00
type: Community
tags:
  - Mentorship
  - AsyncAPI
  - OSS
cover: /img/posts/mentorship-assets/gta.webp
authors:
  - name: Azeez Elegbede
    photo: /img/avatars/ace.webp
    link: https://www.linkedin.com/in/aelegbede/
    byline: AsyncAPI Preacher 
excerpt: 'A Story About Growth, Design, Docs, Code, and Community — The 2024 Mentorship Journey'
---

It doesn't feel like that long ago when we got rejected from participating in GSoC in the summer of 2022, which motivated us to pilot our in-house mentorship program that focuses on encouraging contributors to learn, grow, and contribute to exciting opensource projects while receiving guidance from experienced mentors. 

> Yet, here we are again, wrapping up our third edition of the AsyncAPI Mentorship Program.

Reviewer: ```suggestion
Yet, here we are again, ending our third edition of the AsyncAPI Mentorship Program.
```

As the 2024 mentorship program concludes, it's a perfect time to reflect on what's been an incredibly rewarding journey for everyone involved. Beyond lines of code and completed tasks, this year's program truly focused on collective learning, personal growth, and building something significant as a community.

The 2024 mentorship program was filled with a blend of challenges, surprises and successes. While it's natural to focus on the external events that shaped our experiences, this time of the year invites us as a community to turn inward and ask more profound questions about how we learned and grew. 

## Reflect: Honouring the Lessons of 2024

> For us, **reflection isn't just a simple look back; it's about extracting profound meaning and vital insights from our experiences**. As a community, this process of "meaning-making" is absolutely crucial for our continuous growth and development. It's how we learn, adapt, and build a stronger future together.

Reviewer: ```suggestion
For us, **reflection isn't just a simple look back; it's about extracting profound meaning and vital insights from our experiences**. As a community, the process of "meaning-making" is absolutely crucial for our continuous growth and development. It's how we learn, adapt, and build a stronger future together.
```

In this cohort, we deliberately expanded our project horizons. We didn't just welcome code-related project ideas; we also opened our doors wide to crucial design and technical writing initiatives.

<Figure
  src="/img/posts/mentorship-assets/ideas.webp"
  caption="In rhythm and code, we speak many tongues—diverse in roots, united in purpose."
  className="text-center"
/>

This strategic expansion is a source of immense pride, strengthening the very fabric of AsyncAPI for years to come. This commitment to broader opportunities yielded impressive results:

- **Vibrant Community Involvement:** The 2024 Mentorship Program's kickoff discussion garnered involvement from 39 community members, including maintainers, contributors, and mentors. This vibrant participation led to 13 proposed project ideas, encompassing two design and two technical writing concepts. From these, [eight](https://github.com/orgs/asyncapi/discussions/1361#discussioncomment-10819168) distinct projects were selected, meeting all program requirements.
  

> - **Strong Success Rates:** The 2024 cohort saw a **77.78% mentee success rate** and **90% project completion**. While one coding and one documentation mentee did not pass, the documentation project still achieved full completion thanks to the outstanding collaboration of the remaining mentees.
> 

Reviewer: ```suggestion
- **Strong Success Rates:** This year's cohort had **77.78% mentee success rate** and **90% project completion**. While one coding and one documentation mentee did not pass, the documentation project still managed to be finished thanks to the outstanding collaboration of the remaining mentees.

```

![Mentee outcomes by project type](/img/posts/mentorship-assets/chart.webp)

- **Immediate Value in Action:** At AsyncAPI, the impact of contributing is often felt right away, not just in code, but across all areas of the community. This cohort reflected that beautifully. 
  
  The [Holopin Design Project](https://github.com/asyncapi/community/issues/1306) brought this value to life by creating a community-wide badge to recognize contributors doing great work. It was also used to celebrate all mentees who completed their programs—including the mentee who designed the badge.

- **Improved project diversity:** This cohort not just showed us the impact of diversifying project ideas, but also helps us understand that non-code related project can hold equal values in the community as it's counterpart. 

### Where We Felt Challenged

> It's often through challenges that we gain our most valuable insights. They serve as critical tests of our resilience, adaptability, and resourcefulness, teaching us how to apply these strengths effectively as we move forward.
> 
> The most prominent challenges were the **lack of proper program documentation** and the **vagueness of the program's timeline**.

Reviewer: [suggestion]: the second paragraph dives right into the subsection's topic, so consider starting it with that. 
```suggestion

The most prominent challenges were the **lack of proper program documentation** and the **vagueness of the program's timeline**.
```

> Imagine a scenario where program organizers are the _only_ source of truth. That's precisely what happened without comprehensive documentation. Mentors and mentees found themselves constantly waiting on organizers for basic information, especially when immediate answers weren't possible. This wasn't just inconvenient for them; it was a constant drain on the organizers' time, as they repeated the same details countless times.

Reviewer: ```suggestion
Another challenge that this cohort faced is program organizers were seen as the _only_ source for guidance. Due to the program lacking comprehensive documentation, mentors and mentees found themselves constantly waiting on organizers for basic information, especially when immediate answers weren't possible. This wasn't just inconvenient for them; but for the organizers too, as they repeated the same details countless times.
```

<Figure
  src="/img/posts/mentorship-assets/tired.webp"
  caption="It felt like a never-ending cycle"
  className="text-center"
/>

> Adding to this, the initial decision to keep the program timeline somewhat open-ended inadvertently became a hurdle. Some mentees, interpreting this flexibility as an invitation to take their time, started their projects at a leisurely pace. This slow burn in the early phases meant mentors faced a significant challenge with consistently low output from these projects.

Reviewer: ```suggestion
Adding to this, the initial decision to keep the program's timeline somewhat open-ended inadvertently became a challenge. Some mentees, interpreting this flexibility as an invitation to take their time, started their projects at a leisurely pace. This led to mentors to consistently having low output from these projects.
```

Another unexpected development was the unfortunate reality of mentor withdrawals. While we typically anticipate and prepare for mentees dropping out due to personal reasons, a mentor's early departure was unprecedented for us.

This surprising event served as a powerful reminder: while mentoring is profoundly rewarding, it also presents its own unique challenges and underscores the need for proactive support systems for our mentors.

## Refocus: Setting Our Future Intentions

As we look ahead, we firmly believe that the most significant progress often arises from incremental improvements. We are embracing the concept of **marginal gains** to continually enhance our program. A critical element of this forward movement involves challenging perspectives that no longer serve our evolving goals.

Our assumptions and beliefs profoundly shape our choices, and recognizing their potential limitations has prompted us to thoroughly examine our approach throughout the program's history. This introspection is designed to foster an even more welcoming and inclusive environment.

Some of the changes coming for the next cohort are as follow:

> 1. **A New Identity: The AsyncAPI Maintainership Program.** Moving forward, the AsyncAPI Mentorship Program will be formally renamed to the **AsyncAPI Maintainership Program**. This strategic transition reflects a singular, clear objective: to explicitly guide mentees along the path to becoming active AsyncAPI maintainers. This shift will not only impact the depth and type of contributions but is also designed to significantly increase our pool of active maintainers and triagers within the initiative.
>    
> 2. **Official Program Documentation.** We've learned that relying solely on program organizers as the "single source of truth" for critical information is neither sufficient nor sustainable long-term. To address this, we are introducing comprehensive, official program documentation. This resource will empower all participants with readily accessible information, streamlining processes and reducing reliance on individual organizers.

Reviewer: ```suggestion
1. **A New Identity: Moving forward, the AsyncAPI Mentorship Program will be formally renamed to the **AsyncAPI Maintainership Program**. This strategic transition reflects a singular, clear objective: to explicitly guide mentees along the path to becoming active AsyncAPI maintainers. This shift will not only impact the depth and type of contributions but is also designed to significantly increase our pool of active maintainers and triagers within the initiative.
   
2. **Official Program Documentation.** We've learned that relying solely on program organizers the source of critical information is neither sufficient nor sustainable long-term. To address this, we are introducing comprehensive, official program documentation. This resource will empower all participants with readily accessible information, streamlining processes and reducing reliance on individual organizers.
```
